unit-tests-java8/src/test/java/org/eclipse/collections/test/bag/mutable/sorted/MutableSortedBagNoComparatorTestCase.java

```diff
@@ -42,6 +48,33 @@
         return result;
     }
 
+    @Override
+    @Test
+    default void RichIterable_minByOptional_maxByOptional()
+    {
+        assertEquals(Optional.of("ca"), this.newWith("ed", "da", "ca", "bc", "ab").minByOptional(string -> string.charAt(string.length() - 1)));
+        assertSame(Optional.empty(), this.<String>newWith().minByOptional(string -> string.charAt(string.length() - 1)));
+        assertThrows(NullPointerException.class, () -> this.newWith(new Object[]{null}).minByOptional(Objects::isNull));
+
+        assertEquals(Optional.of("cz"), this.newWith("ew", "dz", "cz", "bx", "ay").maxByOptional(string -> string.charAt(string.length() - 1)));
+        assertSame(Optional.empty(), this.<String>newWith().maxByOptional(string -> string.charAt(string.length() - 1)));
+        assertThrows(NullPointerException.class, () -> this.newWith(new Object[]{null}).maxByOptional(Objects::isNull));
+    }
+
+    @Override
+    @Test
+    default void OrderedIterable_getFirstOptional()
+    {
+        assertEquals(Optional.of(Integer.valueOf(1)), ((OrderedIterable<?>) this.newWith(3, 3, 3, 2, 2, 1)).getFirstOptional());
+    }
+
+    @Override
+    @Test
+    default void OrderedIterable_getLastOptional()
+    {
+        assertEquals(Optional.of(Integer.valueOf(3)), ((OrderedIterable<?>) this.newWith(3, 3, 3, 2, 2, 1)).getLastOptional());
+    }
+
```

Comment:
okay - but I am confused as to why this is in this PR 😕 How is this related to toString() issue from 1196.

Reply:
To be honest, I don't remember how I came across this anymore, because the bugs I found were mostly in Maps, not Bags. However, I wound up stepping through a lot of tests in the debugger to find and fix the bugs, so I probably wound up stepping through a subclass of this TestCase. Once I saw it, it was obviously wrong to me because no assertions should ever be in unit-tests-java8/**/*Test.java. Assertions only go in *TestCase.java.

Reply:
Now that I'm looking around, there are other minor violations of this rule. Specifically, the subclasses of MultiReaderMutableCollectionTestCase are making assertions about classes instead of interfaces. I'm not planning on fixing the problem in general. I just recognized this mistake as one that I wrote so I wanted to fix it myself.

Reply:
Understood. 

Can you split these into separate commits for ease of review and for managing release notes? Thank you!

Reply:
I pushed the tests back down to the subclass / reverted this change. And I rebased on upstream/master.